orbit-love/orbit-browser-extension
# Feature/add actions and additional data

- [x] Moves getOrbitCredentials out of widget and into `background.js`
- [x] Adds "add member to orbit", "view member on orbit" actions
- [x] Adds additional data on Github only
- [x] Moves helper methods to a new helper (widget helper), including stealing some that we want to keep from the old widget
- [x] Fixes.... bug? With number formatter. Limits for 5000 & 1000 were confused, so we could mistakenly show 5000+ when we shouldn't. OR it might have been a reading comprehension problem. Either way is now fixed & tested.

How to test:

1. Update your `manifest.json` to point to the new entrypoint

```
...
  "content_scripts": [
    {
      "run_at": "document_start",
      "matches": ["https://github.com/*"],
      "exclude_matches": ["https://*/login/*"],
      "js": ["widget/githubEntrypoint.js"]
    },
  ]
...
```

2. Visit a github pull request by a member who exists in your workspace. You should see:
   a. Their contirbutions on GitHuib
   b. Their contirbutions on this repo
   c. A link to view their profile on ORbit

<img width="482" alt="CleanShot 2023-05-26 at 16 05 01@2x" src="https://github.com/orbit-love/orbit-browser-extension/assets/45462299/d5645f5c-05b9-481c-95f7-fb14ddedbca8">
   
3. Visit a GH pull request by a member who does not exist on your workspace. You should see:
    a. Their contirbutions on GH globally
    b. A link to add them to GH
  
![CleanShot 2023-05-26 at 16 07 27@2x](https://github.com/orbit-love/orbit-browser-extension/assets/45462299/59051be0-36a6-441c-a48e-b4546dcd0e63)

4. Click the button to add them to GH. This should automatically update to the state of the widget in 2), unless something goes terribly wrong.

![CleanShot 2023-05-26 at 16 08 20@2x](https://github.com/orbit-love/orbit-browser-extension/assets/45462299/7de5c089-c518-4fd5-91a6-4af17f7732e3)
